metadata-service/health-servlet/src/main/java/com/datahub/health/controller/HealthCheckController.java

```diff
@@ -33,6 +35,27 @@ public HealthCheckController() {
         this::getElasticHealth, 5, TimeUnit.SECONDS);
   }
 
+  /**
+   * Combined health check endpoint for checking GMS clients.
+   * For now, just checks the health of the ElasticSearch client
+   * @return A ResponseEntity with a Map of String (component name) to ResponseEntity (the health check status of
+   * that component). The status code will be 200 if all components are okay, and 500 if one or more components are not
+   * healthy.
+   */
+  @GetMapping(path = "/combined", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
I'd prefer something like its intended use case. Like maybe `/ready`

Reply:
I think optional args would be good as well i.e. `/ready?application[0]=elasticsearch&application[1]=mysql` with it defaulting to all if nothing is supplied in the array. Thoughts? Not sure if useful, but I can see someone wanting to configure what applications they want to verify before starting up.

Reply:
I like that idea Ryan, I can implement that